File: Cs50/General Projects/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//prototypes
int letterCount(string in);
int wordCount(string in);
int sentCount(string in);
int CLindex(string in);

int main(void)
{
    string sentence = get_string("Text: ");
    int index = CLindex(sentence);
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}
//count the number of words in a given text
int wordCount(string in)
{
    int count = 1;
    int length = strlen(in);
    for (int i = 0; i < length; i++)
    {
        if (in[i] == ' ')
        {
            count++;
        }
    }
    return count;
}
//count the number of letters in a given text
int letterCount(string in)
{
    int count = 0;
    int length = strlen(in);
    for (int i = 0; i < length; i++)
    {
        if (isalpha(in[i]))
        {
            count++;
        }
    }
    return count;
}
//count the number of sentences in a text
int sentCount(string in)
{
    int count = 0;
    int length = strlen(in);
    for (int i = 0; i < length; i++)
    {
        if (in[i] == '.' || in[i] == '!' || in[i] == '?')
        {
            count++;
        }
    }
    return count;
}

//calculate the coleman-liau index
int CLindex(string in)
{
    int index;
    //average letters per word
    float L = ((float)letterCount(in) / (float)wordCount(in) * 100);
    //average sentences per 100 words
    float S = ((float)sentCount(in) / (float)wordCount(in) * 100);
    //actual calculation
    index = round(0.0588 * L  - 0.296 * S - 15.8);
    return index;
}

File: Cs50/General Projects/pset6/sentimental-readability/readability.py
# RROTUNDU 2023

# Main function
def main():
    sentence = input("Text: ")
    index = CLindex(sentence)
    if (index > 16):
        print("Grade 16+\n")
    elif (index < 1):
        print("Before Grade 1\n")
    else:
        print("Grade " + str(index) + "\n")

# wordCount(String in) function


def wordCount(inString):
    count = 1
    length = len(inString)
    for i in range(length):
        if (inString[i] == ' '):
            count += 1
    return count

# letterCount(String in) function


def letterCount(inString):
    count = 0
    length = len(inString)
    for i in range(length):
        if (inString[i].isalpha()):
            count += 1
    return count

# sentenceCount(String in) function


def sentenceCount(inString):
    count = 0
    length = len(inString)
    for i in range(length):
        if (inString[i] == '.' or inString[i] == '!' or inString[i] == '?'):
            count += 1
    return count

# CLindex(String in) function


def CLindex(inString):
    L = (float(letterCount(inString)) / float(wordCount(inString)) * 100)
    S = (float(sentenceCount(inString)) / float(wordCount(inString)) * 100)
    index = round(0.0588 * L - 0.296 * S - 15.8)
    return index


# main call
main()